patches/api/0480-Leashable-API.patch

```diff
@@ -0,0 +1,143 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Lulu13022002 <41980282+Lulu13022002@users.noreply.github.com>
+Date: Sat, 22 Jun 2024 21:11:58 +0200
+Subject: [PATCH] Leashable API
+
+
+diff --git a/src/main/java/io/papermc/paper/entity/Leashable.java b/src/main/java/io/papermc/paper/entity/Leashable.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..fc847f338642d984e6082a9f32c718a011baffb4
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/entity/Leashable.java
+@@ -0,0 +1,37 @@
++package io.papermc.paper.entity;
++
++import org.bukkit.entity.Entity;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++
++/**
++ * Represents an entity that can be leashed.
++ */
++public interface Leashable extends Entity {
++
++    /**
++     * Returns whether the entity is currently leashed.
++     *
++     * @return whether the entity is leashed
++     */
++    boolean isLeashed();
++
++    /**
++     * Gets the entity that is currently leading this entity.
++     *
++     * @return the entity holding the leash
++     * @throws IllegalStateException if not currently leashed
++     */
++    @NotNull
++    Entity getLeashHolder() throws IllegalStateException;
```

Comment:
What is the gain throwing ISE over null?

Reply:
This is to keep the compatibility with the old method.